src/components/forms/SignUpForm/SignUpForm.tsx

```diff
@@ -0,0 +1,52 @@
+import './SignUpForm.sass';
+import {useForm} from 'react-hook-form';
+import BasicTextField from '@/components/inputs/BasicTextField/BasicTextField';
+import BasicBtn from '@/components/btns/BasicBtn/BasicBtn';
+import {signUpSchema} from '@/constants/signUp.validation';
+import {yupResolver} from '@hookform/resolvers/yup';
+
+interface FormData {
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+  phoneNumber?: string;
+}
+
+const SignUpForm: React.FC = () => {
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: {errors}
+  } = useForm<FormData>({
```

Comment:
You can omit this form data model and use the type of your validation schema. In this case this interface will be unneeded.